lib/base/networking/apis/tumdev/cached_client.dart

```diff
@@ -0,0 +1,159 @@
+import 'dart:io';
+
+import 'package:grpc/grpc_or_grpcweb.dart';
+import 'package:grpc/service_api.dart';
+
+import 'package:flutter/foundation.dart' show kIsWeb;
+import 'package:hive/hive.dart';
+import 'campus_backend.pbgrpc.dart';
+
+class CachedCampusClient extends CampusClient {
+  CachedCampusClient.webCache()
+      : super(_channel(),
+      options: _callOptions(),
+      interceptors: [CacheInterceptor.webCache()]);
+
+  CachedCampusClient.mobileCache(Directory directory)
+      : super(_channel(),
+      options: _callOptions(),
+      interceptors: [CacheInterceptor.mobileCache(directory)]);
+
+  void invalidateCache() {
+    // TODO: implement interceptUnary
+    throw UnimplementedError();
+  }
+
+  static CallOptions _callOptions() {
+    return CallOptions(metadata: {
```

Comment:
Are those parameters fine for the backend? Just out of interest: what's the reasoning behind passing those parameters to the backend?

Reply:
They would be fine, but if they were filled wout, this would be better.